code/HabitTracker/app/src/main/java/com/example/habittracker/activities/eventlist/EventListActivity.java

```diff
@@ -121,62 +180,35 @@ public void onCallbackSuccess(ArrayList<Habit> habitList) {
+                    dm.getAllHabitEvents(SharedInfo.getInstance().getCurrentUser().getUsername(), habitList.get(i), new HabitEventListCallback() {
                         @Override
-                        public void onEvent(@Nullable QuerySnapshot queryDocumentSnapshots, @Nullable
-                                FirebaseFirestoreException error) {
-                            for (QueryDocumentSnapshot doc : queryDocumentSnapshots) {
+                        public void onCallbackSuccess(ArrayList<HabitEvent> eventArrayList) {
+                            for (HabitEvent doc : eventArrayList) {
                                 boolean duplicate_flag = false;
                                 for (int i = 0;i<eventDataList.size();i++) {
-                                    if(eventDataList.get(i).getEventId() == doc.getId()) {
+                                    if(eventDataList.get(i).getEventId() == doc.getEventId()) {
                                         duplicate_flag = true;
-                                        Log.d("same!", String.valueOf(doc.getId()));
                                     }
                                 }
                                 if(duplicate_flag) {
                                     continue;
                                 }
-                                Log.d(DB_TAG, String.valueOf(doc.getData().get("Habit")));
-                                String eventID = doc.getId();
-                                String habitID = (String) doc.getData().get("Habit");
-
-                                Log.d("eventID", String.valueOf(doc.getId()));
-                                Log.d("date", String.valueOf(doc.getData().get("startDate")));
-
-                                ArrayList<Integer> startDate = (ArrayList<Integer>) doc.getData().get("startDate");
-                                String comments = (String) doc.getData().get("comment");
-//                    String location = (String) doc.getData().get("location");
-//                    String image = (String) doc.getData().get("image");
-                                HabitEvent temp = new HabitEvent();
-                                temp.setStartDate(startDate);
-                                temp.setComment(comments);
-                                temp.setEventId(eventID);
-                                temp.setHabit(habitID);
-                                if(!(habitID=="")) {
-                                    eventDataList.add(temp);
+                                if(!(doc.getHabit()=="")) {
+                                    eventDataList.add(doc);
                                 }
 
                             }
                             eventAdapter.notifyDataSetChanged(); // Notifying the adapter to render any new data fetched
                             //from the cloud
                         }
+
+                        @Override
+                        public void onCallbackFailed() {
```

Comment:
Should we have an error message pop up or something?

Reply:
Leave for later.